workspaces/optic-engine/src/interactions/mod.rs

```diff
@@ -72,6 +81,27 @@ pub fn diff(
     .collect()
 }
 
+#[derive(Clone, Debug)]
+pub struct DiffConfig {
+  include_query_params: bool,
+}
+
+impl Default for DiffConfig {
+  fn default() -> Self {
+    Self {
+      include_query_params: false,
+    }
+  }
+}
```

Comment:
Is there a reason we don't want to pass in `include_query_params` as an argument for `default` and instead mutate this via `with_query_params`?

Reply:
Yeah, that's because this is implementing the `Default` trait, which requires `default()` to be callable without any arguments. Hence the builder pattern: let those who don't care go with the default config, while allowing others to configure.